src/popup/components/ConnectWalletForm.tsx

```diff
@@ -37,32 +37,73 @@ export const ConnectWalletForm = ({ publicKey }: ConnectWalletFormProps) => {
     mode: 'onSubmit',
     reValidateMode: 'onBlur',
     defaultValues: {
-      recurring: false
+      recurring: localStorage?.getItem('recurring') === 'true' || false,
+      amount: localStorage?.getItem('amountValue') || undefined,
+      walletAddressUrl: localStorage?.getItem('walletAddressUrl') || undefined
```

Comment:
As the collective JSON object won't be too big, I guess we can store these together? Any rationale to not to that?

Reply:
seemed cleaner than reading, parsing, changing, writing